components/feature/search/src/main/java/mozilla/components/feature/search/widget/VoiceSearchActivity.kt

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.widget
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.speech.RecognizerIntent
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AppCompatActivity
+import java.util.Locale
+
+/**
+ * Launches voice recognition then uses it to start a new web search.
+ */
+abstract class VoiceSearchActivity : AppCompatActivity() {
```

Comment:
Looking at how this is used on Fenix - where a bigger refactoring is needed to rename the current `VoiceSearchActivity` to `FenixVoiceSearchActivity` I propose having a different name in AC with maybe prepending `Base` or `Abstract` to `VoiceSearchActivity`, a naming scheme which is already used which would then allow to have only minimal changes in Fenix.

Reply:
Done